x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/index.tsx

```diff
@@ -269,6 +276,10 @@ export const AgentListPage: React.FunctionComponent<{}> = () => {
               kuery: kuery && kuery !== '' ? kuery : undefined,
               showInactive,
             }),
+            sendGetAgentTags({
```

Comment:
I'm wondering: could it be possible to only fetch the tags once, i.e. all of them and then use filtering to set the filtered tags?

Reply:
This will not be possible because the query used to filter tags is also used in other places. To understand filteredKeery (formerly allTagsKeury) is used to filter agents based on the tags they have on them, but the new allTagsKeury (formerly allAddRemoveTags) must be loaded to get the tags and display them all on the agent's Add Remove Tags popup regardless of the filters. So we still have to send the first kuery to get the filtered list of agents, but also the second kuery to get all existing tags.

Reply:
I understand the point where you are aiming, it is possible to some extent. The filtered kuery still has to be created since it is used in the SearchBar, but it does not have to be sent as REST request to get the agent tags response, since we can use all the existing tags. Please review the changes.